cmd/ocm-backplane/cloud/common.go

```diff
@@ -295,13 +297,51 @@ func (cfg *QueryConfig) getIsolatedCredentials(ocmToken string) (aws.Credentials
 		Credentials: NewStaticCredentialsProvider(seedCredentials.AccessKeyID, seedCredentials.SecretAccessKey, seedCredentials.SessionToken),
 	})
 
-	targetCredentials, err := AssumeRoleSequence(seedClient, assumeRoleArnSessionSequence, cfg.BackplaneConfiguration.ProxyURL, awsutil.DefaultSTSClientProviderFunc)
+	inlinePolicy, err := getTrustedIPInlinePolicy()
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to build inline policy: %w", err)
+	}
+
+	targetCredentials, err := AssumeRoleSequence(
+		seedClient,
+		assumeRoleArnSessionSequence,
+		cfg.BackplaneConfiguration.ProxyURL,
+		awsutil.DefaultSTSClientProviderFunc,
+		&inlinePolicy,
+	)
 	if err != nil {
 		return aws.Credentials{}, fmt.Errorf("failed to assume role sequence: %w", err)
 	}
 	return targetCredentials, nil
 }
 
+func getTrustedIPInlinePolicy() (awsutil.PolicyDocument, error) {
+	IPList, err := ocm.DefaultOCMInterface.GetTrustedIPList()
+	if err != nil {
+		return awsutil.PolicyDocument{}, fmt.Errorf("failed to fetch trusted IP list: %w", err)
+	}
+
+	sourceIPList := []string{}
+	for _, ip := range IPList.Items() {
+		if ip.Enabled() {
+			//This is hack for now to filter only proxy IP's
+			if strings.HasPrefix(ip.ID(), "209.") ||
```

Comment:
I assume your hard-coded ip list is based on the `allowed_ips` in [rh-control-terraform](https://gitlab.cee.redhat.com/service/sre-platform-rhcontrol-terraform/-/blob/master/terraform/backplane-sso-oidc/common/config.tf?ref_type=heads#L22-33). 

However, the prefixes `209.* and 66.*` do not match `allowed_ips` in that repo. (terraform `allowed_ips` has exact IP addresses `"209.132.178.49/32" not all 209.*` which may include unintended addresses.

Also, terraform defines CIDR blocks as `/32 (single IPs)` and your PR assumes `/24` which is an entire range, is that expected ?

Reply:
Good catch .. fixed